### PR TITLE
chore: use errors.Is to check for a specific error

### PR DESCRIPTION
**Pull request by Fish-pro**
Signed-off-by: Fish-pro <zechun.chen@daocloud.io>

Comparing with `==` will fail on wrapped errors, use errors.Is to check for a specific error，select the unmodified part in the following way
```sh
➜  cilium git:(master) ✗ git grep 'err ==' | grep -v ^vendor | sed 's/.*err == \([^ :),;.]\+\).*/\1/g' | grep -i Err | grep -v "err == 0" | grep -v "err == nil" | sort -u
bpf/bpf_sock.c: if (err == -EHOSTUNREACH || err == -ENOMEM) {
pkg/checker/checker_test.go:            c.Assert(equal, check.Equals, err == "")
```

### CODE REVIEW

**Comment by squeed** [comment]
Many of these files are autogenerated -- so these changes will be reverted when you next run `make generate-api` (unless there is a change to swagger I'm not aware of).

**Comment by sayboras** [comment]
As Casey highlighted above, the files are generated from go-swagger tool, which is already with the latest version. 

While it's more aligning with error checking in recent go version, it's unlikely we need to change anytime in cilium (except upgrade to a newer version of go-swagger).

https://github.com/cilium/cilium/blob/8d69c85b2b993b2a14e1b28f332e296151fc4876/Makefile#L28

**Comment by Fish-pro** [comment]
> As Casey highlighted above, the files are generated from go-swagger tool, which is already with the latest version.
> 
> While it's more aligning with error checking in recent go version, it's unlikely we need to change anytime in cilium (except upgrade to a newer version of go-swagger).
> 
> https://github.com/cilium/cilium/blob/8d69c85b2b993b2a14e1b28f332e296151fc4876/Makefile#L28

@sayboras Thanks, which version is a good upgrade for go-swagger?

**Comment by Fish-pro** [comment]
> autogenerated

@squeed Thank you for your attention, I will remove the modification to the auto-generated file, is that OK?

**Comment by Fish-pro** [comment]
Skip `make generate-api` to automatically generate files
```sh
➜  cilium git:(cleanup/erroris) git grep 'err ==' | grep -v ^vendor | grep -v ^api | sed 's/.*err == \([^ :),;.]\+\).*/\1/g' | grep -i Err | grep -v "err == 0" | grep -v "err == nil" | sort -u
bpf/bpf_sock.c:	if (err == -EHOSTUNREACH || err == -ENOMEM) {
pkg/checker/checker_test.go:		c.Assert(equal, check.Equals, err == ""
```

**Review by sayboras — APPROVED** [review]

**Review by nebril — APPROVED** [review]

**Review by tommyp1ckles — APPROVED** [review]

**Review by joamaki — APPROVED** [review]

**Comment by joestringer** [comment]
Sorry, the `master` branch was temporarily broken earlier. Could you rebase once more? We can run CI then.

**Comment by Fish-pro** [comment]
> Sorry, the `master` branch was temporarily broken earlier. Could you rebase once more? We can run CI then.

@joestringer 
Thanks for tips, I have been rebased

**Comment by Fish-pro** [comment]
@joestringer 
Some task errors, could you give me some guidance
image

**Comment by joestringer** [comment]
@Fish-pro that specific failure says that curl attempted to connect to something, but the request timed out. I would need a direct link to the failure to investigate further, since this can depend on which specific workflow failed. Another thing you can try is to search the "issues" page for similar reports, and compare your results vs any previous similar reports.

**Comment by joestringer** [comment]
As for the latest runs, it appears that image builds took too long for several checks which caused the failures. I am not sure why this is, maybe a temporary outage with the docker provider. I will close+reopen the PR to try to re-start the CI.

**Comment by joestringer** [comment]
Please hold off on any further changes or rebase until we get a chance to see the full CI results from the above test command. Thanks!

**Comment by joestringer** [comment]
@Fish-pro was there a reason you just updated the branch again?

**Comment by Fish-pro** [comment]
> This branch is out-of-date with the base branch

@joestringer 
This branch is out-of-date with the base branch

**Comment by joestringer** [comment]
We constantly merge changes into Cilium, so the branch will often be out-of-date. The branch does not need to be 100% up to date with master in order to merge, it just needs to:
(a) Have no conflicts
(b) Have positive reviews from codeowners and no negative reviews
(c) Pass CI

Each time you rebase the PR, it clears the results of (c). If you continue to rebase every day, then we will never merge this PR because we usually do not see the results of CI before you update the PR.

**Comment by joestringer** [comment]
At this time, there was a known issue yesterday with CI so we actually do need you to rebase this time just to pick up the latest CI fixes.

**Comment by joestringer** [comment]
Please do not update the PR for now, so we can see the results of the test.

**Comment by aanm** [comment]
The results were all green.

**Comment by pchaigno** [comment]
Reviews are in and tests are passing. Merging.

**Pull request merged**
